packages/controller-utils/src/util.test.ts

```diff
@@ -611,3 +611,29 @@ describe('util', () => {
     });
   });
 });
+
+describe('isEqualCaseInsensitive', () => {
+  it('returns false for non-string values', () => {
+    // @ts-expect-error Invalid type for testing purposes
+    expect(util.isEqualCaseInsensitive(null, 'test')).toBe(false);
```

Comment:
Nit: Maybe useful to have examples where the inputs _are_ equal, but it still returns `false` (because they're not strings)

```suggestion
    // @ts-expect-error Invalid type for testing purposes
    expect(util.isEqualCaseInsensitive(null, null)).toBe(false);
    // @ts-expect-error Invalid type for testing purposes
    expect(util.isEqualCaseInsensitive(5, 5)).toBe(false);
    // @ts-expect-error Invalid type for testing purposes
    expect(util.isEqualCaseInsensitive(null, 'test')).toBe(false);
```